go/pkg/reader/reader.go

```diff
@@ -209,33 +209,34 @@ func (cfs *compressedSeeker) Read(p []byte) (int, error) {
 	// to plan" error. EOF implies our uncompressed input data has finished reading,
 	// but it doesn't mean that we've finished preparing/returning our compressed data.
 	// We only want to return EOF to the user when the _compressed_ data is finished reading.
+	var finalErr error
 	if errR != nil && errR != io.EOF {
-		cfs.err = errR
+		finalErr = errR
 	} else if errW != nil {
-		cfs.err = errW
+		finalErr = errW
 	}
 
 	// When the buffer ends, or in case of _any_ error, we compress all the bytes we
 	// had available. The encoder requires a Close call to finish writing compressed
 	// data smaller than zstd's window size.
 	var errC error
-	if cfs.err != nil || errR == io.EOF {
+	if finalErr != nil || errR == io.EOF {
 		errC = cfs.encdW.Encoder.Close()
```

Comment:
why bother with errC at all? can this just assign directly to finalErr?

Reply:
Done.